### PR TITLE
PPU Precise: FPCC handling on float ops implemented

### PR DESCRIPTION
**Pull request by elad335**
This adds missing FPCC tests on floting-point ops for ppu precise, and optimizes existing tests for ppu interpreters.

Also improve error checking of sys_semaphore_create: Return EINVAL on SYS_SYNC_PRIORITY_INHERIT protocol.

### CODE REVIEW

**Comment by uaqlover** [comment]
it been huge improvement with your pull requests thanks lot

**Comment by jobs-git** [comment]
please solve conflicts

**Comment by elad335** [comment]
Updated, cleaned up a few things.

File: rpcs3/Emu/Cell/PPUInterpreter.cpp [hunk 1]
```diff
@@ -4688,11 +4770,25 @@ bool ppu_interpreter::MTFSFI(ppu_thread& ppu, ppu_opcode_t op)
 	}
 	else
 	{
-		u32 i = op.i;
-		ppu.fpscr.bits[bf + 3] = i & 1; i >>= 1;
-		ppu.fpscr.bits[bf + 2] = i & 1; i >>= 1;
-		ppu.fpscr.bits[bf + 1] = i & 1; i >>= 1;
-		ppu.fpscr.bits[bf + 0] = i & 1;
+		// TODO: Constexpr
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/PPUInterpreter.cpp]: Just don't use std::invoke to call lambda, and it should compile as static constexpr.

**Pull request merged**